demo/client/src/components/addModel.js

```diff
@@ -67,10 +67,13 @@ class AddModel extends React.Component {
   constructor(props) {
     super(props);
 
-    this.storageFactory = new DataStoreFactory();
-
     // Default to local storage for storing original data.
     const storageType = localStorage.getItem('storageType') || 'local';
+    const storageFactory = new DataStoreFactory();
+    this.storages = {
```

Comment:
Maybe this can be moved to a global place so that it does not have to get re-initialized.